access/resource_ip_access_list.go

```diff
@@ -31,47 +32,77 @@ func ResourceIPAccessList() *schema.Resource {
 	return common.Resource{
 		Schema: s,
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			w, err := c.WorkspaceClient()
-			if err != nil {
-				return err
-			}
 			var iacl settings.CreateIpAccessList
+			var updateIacl settings.UpdateIpAccessList
 			common.DataToStructPointer(d, s, &iacl)
-			status, err := w.IpAccessLists.Create(ctx, iacl)
-			if err != nil {
-				return err
-			}
-			d.SetId(status.IpAccessList.ListId)
-			return nil
+			common.DataToStructPointer(d, s, &updateIacl)
+			return c.AccountOrWorkspaceRequest(func(acc *databricks.AccountClient) error {
+				status, err := acc.IpAccessLists.Create(ctx, iacl)
+				if err != nil {
+					return err
+				}
+				//need to enable the IP Access List with update
+				if d.Get("enabled").(bool) {
+					updateIacl.IpAccessListId = status.IpAccessList.ListId
+					err = acc.IpAccessLists.Update(ctx, updateIacl)
+					if err != nil {
+						return err
+					}
+				}
+				d.SetId(status.IpAccessList.ListId)
+				return nil
+			}, func(w *databricks.WorkspaceClient) error {
+				status, err := w.IpAccessLists.Create(ctx, iacl)
+				if err != nil {
+					return err
+				}
+				d.SetId(status.IpAccessList.ListId)
+				return nil
+			})
 		},
 		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			w, err := c.WorkspaceClient()
-			if err != nil {
-				return err
-			}
-			status, err := w.IpAccessLists.GetByIpAccessListId(ctx, d.Id())
-			if err != nil {
-				return err
-			}
-			common.StructToData(status.IpAccessList, s, d)
-			return nil
+			return c.AccountOrWorkspaceRequest(func(acc *databricks.AccountClient) error {
+				_, err := acc.IpAccessLists.GetByIpAccessListId(ctx, d.Id())
+				if err != nil {
+					return err
+				}
```

Comment:
If it returns blank response, do we need this piece of code?

Reply:
this is to check if the access list id exists, before listing all the access lists and find the right one